internal/config.go

```diff
@@ -177,6 +185,10 @@ func mergeAndValidateOIDCConfigs(cfg *configv1.Config) error {
 }
 
 func applyOIDCDefaults(config *oidcv1.OIDCConfig) {
+	if config.GetClientAuthenticationMethod() == "" {
+		config.ClientAuthenticationMethod = ClientAuthenticationBasic
+	}
```

Comment:
For backwards compatibility, we allow the field to not be set, but default it to basic, to preserve behaviour.